packages/ember-metal/lib/expand_properties.js

```diff
@@ -41,6 +62,9 @@ export default function expandProperties(pattern, callback) {
     'Brace expanded properties cannot contain spaces, e.g. "user.{firstName, lastName}" should be "user.{firstName,lastName}"',
     pattern.indexOf(' ') === -1
   );
+  assert(
+    'Brace expanded properties have to be balanced and cannot be nested, e.g. "user.{name.{first,last},age}" and "user.{firstName},lastName}" are not allowed',
```

Comment:
Can you tweak the message to show the actual brace expansion that the user used (instead of a static value)?  Possibly do the same for the assertion just above?